code/clue.js

```diff
@@ -1,59 +1,548 @@
+  return array[Math.floor(Math.random() * array.length)];
+};
 
-const weapons = []
+////////////////////////////////////////////////////////////////////////////////////////
 
-const rooms = []
+// *** tryed to solve and undertand this problem in sandbox:
+//https://codesandbox.io/s/random-pet-with-favourite-pet-u11le *** WITH COMMENTS!
 
-// THIS FUNCTION WILL RANDOMLY SELECT ONE ITEM FROM THE ARRAY THAT YOU PASS IN TO THE FUNCTION.
-// YOU DON'T NEED TO CHANGE THIS, JUST TRY TO UNDERSTAND IT. AND HOW TO USE IT.
-const randomSelector = array => {
-  return array[Math.floor(Math.random() * array.length)]
-}
+// *** Most-likely machine ***
+const mostLikelyRandomSelector = (items, favouriteItem) => {
+  // For each favouriteItem there will be space for, in this case, 7 more of them in an array. E.g. "7 knifes" or "7 kitchens".
+  const favouriteItemsDuplicated = new Array(7).fill(favouriteItem);
 
-// CREATE AN OBJECT THAT KEEPS THE MYSTERY.
-// With a killer, a weapon and a room.
-// The values will be set later.
+  // (...spread) Takes the array and spreads the items out.
+  // E.g. taking out all your nested individual russian dolls and lay them out in its own individual pieces.
+  // Explains Spread ---> https://www.samanthaming.com/tidbits/92-6-use-cases-of-spread-with-array/
+  // Merging arrays. RandomArray = random Array + favouriteItemsDuplicated (e.g all weapons + 7 knifes):
+  const randomArray = [...items, ...favouriteItemsDuplicated];
+
+  // Taking out a random index from the randomArray (items + favouriteItemsDuplicated)
+  const randomIndex = Math.floor(Math.random() * randomArray.length);
+
+  // Returning the randomIndex ---> [] from the randomArray
+  return randomArray[randomIndex];
+};
+
+////////////////////////////////////////////////////////////////////////////////////////
+
+// *** Clicking "KILLER CARD" ***
 
-// This function will be invoked when you click on the killer card.
 const pickKiller = () => {
-  // This will randomly select a killer from the suspects. And add that to the mystery object.
-  mystery.killer = randomSelector(suspects)
+  // flips card
+  document.getElementById("killerCard").classList.add("flip");
+
+  // time delay starts
+  setTimeout(() => {
+    // random killer selector
+    mystery.killer = randomSelector(suspects);
+
+    // *** changing card info ***
+
+    // card background color
+    document.getElementById("killerCard").style.background = mystery.killer.color;
+    // heading background color
+    document.getElementById("cardHeadingKiller").style.background = mystery.killer.heading;
+    // killer name
+    document.getElementById("killerName").innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`;
+    // killer occupation
+    document.getElementById("killerOccupation").innerHTML = `${mystery.killer.occupation}`;
+    //killer image
+    document.getElementById("killerImage").src = `${mystery.killer.image}`;
+
+    // time delay millisecond
+  }, 300);
```

Comment:
Small detail, the image of the card changes after the animation.  To make it more seamless, how would you change the image halfway through the animation?